backend/app.py

```diff
@@ -8,42 +8,45 @@
 import os
 
 
-def main():
+def index():
     if not os.path.exists(pdb_data_path):
-        log('Downloading data.')
+        log("Downloading data.")
         download_csv(pdb_data_url, pdb_data_path)
-        log('Culling PDB ID duplicates.')
+        log("Culling PDB ID duplicates.")
         cull_duplicates(pdb_data_path)
 
-    log('Converting protein data to DocumentArray')
+    log("Converting protein data to DocumentArray")
     with open(pdb_data_path) as data_file:
         docs_generator = from_csv(
-            fp=data_file,
-            field_resolver={
-                "sequence": "text",
-                "structureId": "id"
-            }
+            fp=data_file, field_resolver={"sequence": "text", "structureId": "id"}
         )
         proteins = DocumentArray(docs_generator)
     log(f"Loaded {len(proteins)} proteins from {pdb_data_path}.")
 
-    log('Creating flow.')
+    log("Building index.")
+    indexer = Flow(protocol="grpc").add(uses=ProtBertExecutor).add(uses=MyIndexer)
+    with indexer:
+        indexer.index(proteins, request_size=10)
+
+
+def main():
+
+    if not os.path.exists(embeddings_path):
+        log("Embeddings not found.")
+        index()
+
```

Comment:
```suggestion
    if not os.path.exists(embeddings_path):
        index()
    else:
        log(f"Embeddings found {embeddings_path}. Skipping index step. Remove this file and re-run to index.")
       
``

Reply:
The indexing step can be completely independent. At a certain point we can have a script to run from CLI for indexing.